test_scripts/Defects/4_5/1892_HB_from_SDL_after_receiving_HB_from_mobile.lua

```diff
@@ -0,0 +1,147 @@
+  self.mobileSession1.ignoreHeartBeatAck = false
+  self.mobileSession1:StartService(7)
+end
+
+local function RegisterAppInterface(self)
+  local CorIdRegister = self.mobileSession1:SendRPC("RegisterAppInterface", {
+      syncMsgVersion ={
+        majorVersion = 4,
+        minorVersion = 3
+      },
+      appName = config.application1.registerAppInterfaceParams.appName,
+      isMediaApplication = true,
+      languageDesired = 'EN-US',
+      hmiDisplayLanguageDesired = 'EN-US',
+      appHMIType = { "DEFAULT" },
+      appID = config.application1.registerAppInterfaceParams.appID
+    })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+    { application = {appName = config.application1.registerAppInterfaceParams.appName }})
+  self.mobileSession1:ExpectResponse(CorIdRegister, { success = true, resultCode = "SUCCESS" })
+  self.mobileSession1:ExpectNotification("OnHMIStatus",
+    { systemContext = "MAIN", hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE" })
+end
+
+local function ExpectationAfterAppRegistration(self)
+  local hmiAppId = commonDefects.getHMIAppId(1)
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered",
+    {unexpectedDisconnect = true, appID = hmiAppId})
+  :Times(0)
+  commonFunctions:userPrint (33,"Log: AppSession started, HB disabled")
+  commonFunctions:userPrint (33, "Log: App v.3 disconnection not expected since no HB ACK and timer" ..
+  "should be started by SDL till the HB request from app first")
+  local HBEvent = events.Event()
+  HBEvent.matches =
+  function(_, data)
+    return data.frameType == 0 and
+    data.serviceType == 0 and
+    (data.sessionId == self.mobileSession1.sessionId) and
+    data.frameInfo == 0
+  end
+  self.mobileSession1:ExpectEvent(HBEvent, "HB")
+  :Times(0)
+  commonDefects.delayedExp(10000)
+end
+
+local function sendHBFromMobileAndReceivingFromSDL(self)
+  self.mobileSession1:Send({
+      frameType = constants.FRAME_TYPE.CONTROL_FRAME,
+      serviceType = constants.SERVICE_TYPE.CONTROL,
+      frameInfo = constants.FRAME_INFO.HEARTBEAT
+    })
+  local HBEvent = events.Event()
+  HBEvent.matches =
+  function(_, data)
+    return data.frameType == 0 and
+    data.serviceType == 0 and
+    (data.sessionId == self.mobileSession1.sessionId) and
+    data.frameInfo == 0
+  end
+  self.mobileSession1:ExpectEvent(HBEvent, "HB")
+  local HBACKEvent = events.Event()
+  HBACKEvent.matches =
+  function(_, data)
+    return data.frameType == 0 and
+    data.serviceType == 0 and
+    (data.sessionId == self.mobileSession1.sessionId) and
+    data.frameInfo == 255
+  end
+  self.mobileSession1:ExpectEvent(HBACKEvent, "HB")
+end
+
+local function DisconnectDueToHeartbeat(self)
+  local hmiAppId = commonDefects.getHMIAppId(1)
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", {unexpectedDisconnect = true, appID = hmiAppId})
+  commonDefects.delayedExp()
+  commonFunctions:userPrint(33, "AppSession3 started, HB enabled", self.mobileSession3)
```

Comment:
@GetmanetsIrina what is `mobileSession3`? Shouldn't you remove it?

Reply:
Removed.